akka-stream/src/main/scala/akka/stream/javadsl/package.scala

```diff
@@ -6,6 +6,8 @@ package akka.stream
 package object javadsl {
   def combinerToScala[M1, M2, M](f: akka.japi.function.Function2[M1, M2, M]): (M1, M2) ⇒ M =
     f match {
+      case x if x eq Keep.left   ⇒ scaladsl.Keep.left.asInstanceOf[(M1, M2) ⇒ M]
```

Comment:
In theory, would http://www.scala-lang.org/api/current/#scala.annotation.unchecked.uncheckedStable do the trick?